x/ibc/03-connection/client/cli/tx.go

```diff
@@ -1,298 +1,474 @@
-			time.Sleep(8 * time.Second)
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
+		},
+	}
+	return cmd
+}
 
-			header, err = getHeader(ctx1)
-			if err != nil {
-				return err
-			}
+// GetCmdConnectionOpenConfirm defines the command to initialize a connection on
+// chain A with a given counterparty chain B
+func GetCmdConnectionOpenConfirm(storeKey string, cdc *codec.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "open-confirm [connection-id] [path/to/proof_ack.json]",
+		Short: "confirm to chain B that connection is open on chain A",
+		Args:  cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			txBldr := auth.NewTxBuilderFromCLI().WithTxEncoder(utils.GetTxEncoder(cdc))
+			cliCtx := context.NewCLIContext().
+				WithCodec(cdc).
+				WithHeight(viper.GetInt64(flags.FlagHeight))
+			// cliCtx2 := context.NewCLIContextWithFrom(viper.GetString(FlagFrom2)).
+			// WithCodec(cdc).
+			// WithNodeURI(viper.GetString(FlagNode2)).
+			// WithBroadcastMode(flags.BroadcastBlock)
 
-			msgUpdate = client.MsgUpdateClient{
-				ClientID: conn2.Client,
-				Header:   header,
-				Signer:   ctx2.GetFromAddress(),
-			}
+			connectionID := args[0]
 
-			err = utils.GenerateOrBroadcastMsgs(ctx2, txBldr, []sdk.Msg{msgUpdate})
+			proofBz, err := ioutil.ReadFile(args[1])
 			if err != nil {
 				return err
 			}
 
-			q1 = storestate.NewCLIQuerier(ctx1.WithHeight(header.Height - 1))
-
-			_, pstate, err = obj1.StageCLI(q1)
-			if err != nil {
+			var proofAck ics23.Proof
+			if err := cdc.UnmarshalJSON(proofBz, &proofAck); err != nil {
 				return err
 			}
 
-			msgConfirm := connection.MsgOpenConfirm{
-				ConnectionID: connId2,
-				Proofs:       []commitment.Proof{pstate},
-				Height:       uint64(header.Height),
-				Signer:       ctx2.GetFromAddress(),
-			}
+			proofHeight := uint64(cliCtx.Height)
 
-			err = utils.GenerateOrBroadcastMsgs(ctx2, txBldr, []sdk.Msg{msgConfirm})
-			if err != nil {
-				return err
-			}
+			msg := types.NewMsgConnectionOpenConfirm(
+				connectionID, proofAck, proofHeight, cliCtx.GetFromAddress(),
+			)
 
-			return nil
+			return utils.GenerateOrBroadcastMsgs(cliCtx, txBldr, []sdk.Msg{msg})
 		},
 	}
+	return cmd
+}
 
-	// TODO: Provide flag description
-	cmd.Flags().String(FlagNode1, "tcp://localhost:26657", "")
-	cmd.Flags().String(FlagNode2, "tcp://localhost:26657", "")
-	cmd.Flags().String(FlagFrom1, "", "")
-	cmd.Flags().String(FlagFrom2, "", "")
+// lastHeight util function to get the consensus height from the node
+func lastHeight(cliCtx context.CLIContext) (uint64, error) {
+	node, err := cliCtx.GetNode()
+	if err != nil {
+		return 0, err
+	}
 
-	cmd.MarkFlagRequired(FlagFrom1)
-	cmd.MarkFlagRequired(FlagFrom2)
+	info, err := node.ABCIInfo()
+	if err != nil {
+		return 0, err
+	}
 
-	return cmd
+	return uint64(info.Response.LastBlockHeight), nil
 }
+
+// func GetCmdHandshake(storeKey string, cdc *codec.Codec) *cobra.Command {
```

Comment:
We need to keep this command in here. I'm refactoring it in the integration PR...

Reply:
I have this command properly implemented in the IBC integration branch. Lets just move it from there to here before we merge this.